test/performance/search/search_benchmark.cpp

```diff
@@ -162,8 +162,12 @@ void unidirectional_search_all_collection(benchmark::State & state, options && o
     seqan3::fm_index index{collection};
     seqan3::configuration cfg = seqan3::search_cfg::max_error_total{seqan3::search_cfg::error_count{o.searched_errors}};
 
+    size_t sum{};
     for (auto _ : state)
+    {
         auto results = search(reads, index, cfg);
+        benchmark::DoNotOptimize(sum += std::ranges::distance(results));
+    }
 }
```

Comment:
I think we should either do

```cpp
    size_t sum{};
    for (auto _ : state)
    {
        auto results = search(reads, index, cfg);
        sum += std::ranges::distance(results);
    }
    benchmark::DoNotOptimize(sum);
```

or 
```cpp
    for (auto _ : state)
    {
        benchmark::DoNotOptimize(search(reads, index, cfg));
    }
```

Do these make a difference for you?

Reply:
I assume that the sum is only needed to use the results, so the compiler does not discard the whole search?
Please add a comment on this or use the second suggestion (if it works).
Maybe you can also write `benchmark::DoNotOptimize(results);`, which is clearer than using the sum?

Reply:
Yes, `benchmark::DoNotOptimize(results);` is also fine. I think the main/first question is whether we want `benchmark::DoNotOptimize` inside or outside the loop.

It's probably more clear if it is outside the loop and may cause less worrying about whether this is now measured currently or not.

Generally, we can kinda do whatever we want as long as we use the macro:

> `DoNotOptimize(<expr>)` forces the result of `<expr>` to be stored in either memory or a register. For GNU based compilers it acts as read/write barrier for global memory. More specifically, it forces the compiler to flush pending writes to memory and reload any other values as necessary.
> 
> Note that `DoNotOptimize(<expr>)` does not prevent optimizations on `<expr>` in any way. `<expr>` may even be removed entirely when the result is already known.